### PR TITLE
--Scene Dataset Config paths glob/wildcard handling

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
This PR adds the ability to provide glob wildcards in the scene dataset configuration for the paths to specific configs, enabling wild-card-based searches and loads for configs that may be spread across many directories, such as per-stage paths that exist in Mp3d and HM3D datsets.  See the newly added config file data/test_assets/dataset_tests/dataset_1/test_dataset_1.scene_dataset_config.json for usage examples.

## How Has This Been Tested
C++ and python tests pass

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/io/io.h [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #ifndef ESP_IO_IO_H_
 #define ESP_IO_IO_H_
 
+#include <glob.h>
```
Comment (mosra) [inline src/esp/io/io.h:8]: Can be in the cpp instead :)

File: src/esp/io/io.h [hunk 2]
```diff
@@ -19,6 +20,14 @@ std::string removeExtension(const std::string& file);
 
 std::string changeExtension(const std::string& file, const std::string& ext);
 
+/**
+ * @brief This function will perform globbing on a passed path + pattern,
+ * returning all the files and directories that match the pattern.
+ * @param pattern The pattern to match
```
Comment (mosra) [inline src/esp/io/io.h:26]: Maybe mention what is allowed in the pattern? As in, it's just wildcards and not full regular expressions.

**Review by mosra — COMMENTED** [review]
The glob() API usage is simpler than I expected, nice.

File: src/esp/io/io.h (hunk 2 above)
Comment (jturner65) [reply src/esp/io/io.h:25]: Is there a way in the docs to put an external html link? (I was thinking of linking wiki :D )

Comment (mosra) [reply src/esp/io/io.h:25]: It's a standard Markdown, so `[title](url)` works. Or just put an URL there directly.

**Review by Skylion007 — APPROVED** [review]
Looks pretty good to me.

**Pull request merged**